src/Extensions/Promise.test.ts

```diff
@@ -40,4 +41,18 @@ describe('Promise', async (should: any) => {
       expected: ERROR,
     })
   }
+
+  {
+    const actual = await Promise.reject(10)
+      .rethrow(error => error === 10 && TRANSLATION)
```

Comment:
Maybe add a ternary or `if` in the translate callback to make the intended behavior more explicit?

```
rethrow(error => error === 10 ? TRANSLATION : 'Something unexpected happened')
```